pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,341 @@
+
+    def _count_projects(
+        self,
+        namespace: str,
+        search_str: str = "",
+        admin: Union[str, List[str]] = False,
+    ) -> int:
+        """
+        Get total number of found projects. [This function is related to _find_projects]
+        :param namespace: namespace where to search for a project
+        :param search_str: search string. will be searched in name, tag and description information
+        :param admin: True, if user is admin for this namespace
+        :return: number of found project in specified namespace
+        """
+        search_str = f"%%{search_str}%%"
+        admin_tuple = tuple_converter(admin)
+        if namespace:
+            and_namespace_sql = f"""AND {NAMESPACE_COL} = %s"""
+            namespace = (namespace,)
+        else:
+            and_namespace_sql = ""
+            namespace = tuple()
+
+        count_sql = f"""
+        select count(*)
+            from {DB_TABLE_NAME} 
+                where ({NAME_COL} ILIKE %s or ({PROJ_COL}->>'description') ILIKE %s or {TAG_COL} ILIKE %s)
+                    and ({PRIVATE_COL} is %s or {NAMESPACE_COL} in %s ) {and_namespace_sql};"""
+        result = self.con.run_sql_fetchall(
+            count_sql,
+            search_str,
+            search_str,
+            search_str,
+            False,
+            admin_tuple,
+            *namespace,
+        )
+        try:
+            number_of_prj = result[0][0]
+        except IndexError:
+            number_of_prj = 0
+        return number_of_prj
+
+    def _get_projects(
+        self,
+        namespace: str = None,
+        search_str: str = "",
+        admin: Union[str, List[str]] = False,
+        limit: int = DEFAULT_LIMIT,
+        offset: int = DEFAULT_OFFSET,
+    ) -> List[AnnotationModel]:
+        """
+        Search for project inside namespace by providing search string.
+        :param namespace: namespace where to search for a project
+        :param search_str: search string that has to be found in the name, tag or project description
+        :param admin: True, if user is admin of the namespace [Default: False]
+        :param limit: limit of return results
+        :param offset: number of results off set (that were already showed)
+        :return: list of found projects with their annotations.
+        """
+        search_str = f"%%{search_str}%%"
+        admin_tuple = tuple_converter(admin)
+
+        if namespace:
+            and_namespace_sql = f"""AND {NAMESPACE_COL} = %s"""
+            namespace = (namespace,)
+        else:
+            and_namespace_sql = ""
+            namespace = tuple()
+
+        count_sql = f"""
+        select {NAMESPACE_COL}, {NAME_COL}, {TAG_COL}, {N_SAMPLES_COL}, ({PROJ_COL}->>'description'), {DIGEST_COL}, {PRIVATE_COL}, {SUBMISSION_DATE_COL}, {LAST_UPDATE_DATE_COL}
+            from {DB_TABLE_NAME} 
+                where ({NAME_COL} ILIKE %s or ({PROJ_COL}->>'description') ILIKE %s or {TAG_COL} ILIKE %s) 
```

Comment:
Same thing here -- `LIKE` should be used ONLY when a search string is explicitly provided